main.go

```diff
@@ -60,25 +60,20 @@ func main() {
 		printErrorFatal("Invalid Usage", err)
 	}
 
-	if len(ctx.Args) > 0 {
-		switch ctx.Args[0] {
-
-		case "version":
-			if Version == "" {
-				if info, ok := debug.ReadBuildInfo(); ok && info.Main.Sum != "" {
-					Version = info.Main.Version
-				} else {
-					Version = "unknown (built from source)"
-				}
-			}
-			version := fmt.Sprintf("freeze version %s", Version)
-			if len(CommitSHA) >= shaLen {
-				version += " (" + CommitSHA[:shaLen] + ")"
+	if config.Version {
+		if Version == "" {
+			if info, ok := debug.ReadBuildInfo(); ok && (info.Main.Version != "" && info.Main.Version != "(devel)") {
+				Version = info.Main.Version
+			} else {
+				Version = "latest (built from source)"
 			}
-
-			fmt.Println(version)
-			os.Exit(0)
 		}
+		version := fmt.Sprintf("freeze version: %s", Version)
+		if len(CommitSHA) > 7 {
+			version += fmt.Sprintf(" (%s)", CommitSHA[:shaLen])
+		}
+		fmt.Println(version)
+		os.Exit(0)
```

Comment:
This zero exit code here is strange to me

I know you are only refactoring but using is Exit leads to code that cannot be tested without mocking.

Won't a `return nil` would have provided the same result?

Reply:
Thanks for doing  the changes, it's safer like this, and can be tested if needed